src/OrchardCore/OrchardCore.ResourceManagement/ServiceCollectionExtensions.cs

```diff
@@ -14,4 +15,14 @@ public static IServiceCollection AddResourceManagement(this IServiceCollection s
 
         return services;
     }
+
+    /// <summary>
+    /// Adds a service implementing <see cref="IConfigureOptions{ResourceManagementOptions}"/> to the service collection.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type of the service implementing <see cref="IConfigureOptions{ResourceManagementOptions}"/> to.
+    /// </typeparam>
+    public static IServiceCollection AddResourceManagementOptionsConfiguration<T>(this IServiceCollection services)
```

Comment:
```suggestion
    public static IServiceCollection ConfigureResourceManagementOptions<T>(this IServiceCollection services)
```

Reply:
Just one suggestion for these is enough for me to get the idea ;). However, I disagree: all of these shortcuts we have are named `Add*`, and for a reason: we're adding a type to the service collection, following methods like `AddTransient`. The `Configure*` methods deal directly with options classes, e.g.:

```c#
services.Configure<TemplateOptions>(o =>
{
    o.MemberAccessStrategy.Register<DisplayMediaFieldViewModel>();
    o.MemberAccessStrategy.Register<Anchor>();

    o.Filters.AddFilter("img_tag", MediaFilters.ImgTag);
})
```

This rather adds a service that implements `IConfigureOptions<T>`, only which in turn deals with options classes.

Reply:
Internally you configure `ResourceManifest` by adding an entry on each call

Reply:
Yeah, but that's not what this method is doing. If we were to swap the method in this:

```c#
services.Configure<ResourceManagementOptions>(o =>
{
    _manifest = new ResourceManifest();
    _manifest
        .DefineScript("coming-soon")
        .SetUrl("~/TheComingSoonTheme/js/scripts.min.js", "TheComingSoonTheme/js/scripts.js")
        .SetVersion("6.0.5");

    _manifest
        .DefineStyle("coming-soon")
        .SetUrl("~/TheComingSoonTheme/css/styles.min.css", "TheComingSoonTheme/css/styles.css")
        .SetVersion("6.0.5");

    o.ResourceManifests.Add(_manifest);
})
```

Then we could name it `ConfigureResourceManagementOptions` and thus do:

```c#
services.ConfigureResourceManagementOptions(o =>
{
   ...
    o.ResourceManifests.Add(_manifest);
})
```

There the name would be appropriate. However, not for the `AddTransient<IConfigureOptions<ResourceManagementOptions>, T>()` substitution I implemented.